forge-game/src/main/java/forge/game/cost/CostPartWithList.java

```diff
@@ -136,9 +140,17 @@ protected boolean executePayment(Player payer, SpellAbility ability, CardCollect
             }
             cardList.addAll(doListPayment(payer, ability, targetCards, effect));
         } else {
+            CardCollection tapped = new CardCollection();
```

Comment:
you could try to implement `canPayListAtOnce()` for `CostTapType`
then this can moved there for cleaner structure

Reply:
Seems to work!